File: benchmarks/C/src/matrix_multiplication.c
#include "matrix_multiplication.h"

void matmul(int A[][N], int B[][N], int R[][N]) {
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            R[i][j] = 0;
            for (int k = 0; k < N; ++k)
                R[i][j] += A[i][k] * B[k][j];
        }
    }
}

File: benchmarks/Python++/Numba/matrix_multiplication.py
from numba import jit

@jit(nopython=True)
def matmul(A, B, R):
    for i in range(len(A)):
        for j in range(len(B)):
            for k in range(len(B)):
                R[i][j] += A[i][k] * B[k][j]